pombola/south_africa/tests/test_elections.py

```diff
@@ -0,0 +1,83 @@
+from django.core.urlresolvers import reverse
+from django.test import TestCase
+from nose.plugins.attrib import attr
+
+from django_date_extensions.fields import ApproximateDate, ApproximateDateField
+from info.models import InfoPage
+from mock import Mock
+from pombola.core.models import (Organisation, OrganisationKind, Person, Place,
+                                 PlaceKind, Position, PositionTitle)
+from pombola.south_africa.views.elections import get_candidate_ranking_sort_key
+
+
+@attr(country="south_africa")
+class ProvincialElectionViewTest(TestCase):
+    def setUp(self):
+        self.province_name = "north-west"
+        provincial_list_slug = "-provincial-%s-election-list-2019" % self.province_name
+
+        self.province_kind = PlaceKind.objects.create(name="Province", slug="province",)
+        self.province = Place.objects.create(
+            kind=self.province_kind, slug=self.province_name, name=self.province_name
+        )
+        self.party_kind = OrganisationKind.objects.create(name="Party", slug="party",)
+        self.election_list_kind = OrganisationKind.objects.create(
+            name="Election List", slug="election-list",
+        )
+        self.party_da = Organisation.objects.create(
+            slug="da", name="DA", kind=self.party_kind
+        )
+
+        self.election_list_da = Organisation.objects.create(
+            slug=(self.party_da.slug + provincial_list_slug),
+            name="DA Election List",
+            kind=self.election_list_kind,
+        )
+
+        self.candidate_a = Person.objects.create(
+            legal_name="Tom Jones", slug="tom-jones"
+        )
+
+        position = Position.objects.create(
+            person=self.candidate_a,
+            organisation=self.election_list_da,
+            start_date=ApproximateDate(2016),
+            end_date=ApproximateDate(2020),
+            title=PositionTitle.objects.create(
+                name="1st Candidate", slug="1st-candidate"
+            ),
+        )
+
+    def test_sa_election_province_candidates_view(self):
+        url = reverse(
+            "sa-election-candidates-provincial", args=[2019, self.province_name]
+        )
+        response = self.client.get(url)
+        self.assertIn(self.candidate_a.name, response.content)
+
+    def test_sa_election_province_party_candidates_view(self):
+        url = reverse(
+            "sa-election-candidates-provincial-party",
+            args=[2019, self.province_name, "da"],
+        )
+        response = self.client.get(url)
+        self.assertIn(self.candidate_a.name, response.content)
+
+
+@attr(country="south_africa")
+class TestGetCandidateRankingSortKey(TestCase):
+    def test_get_ranking_from_number(self):
+        candidate_a = Mock()
```

Comment:
Please use the actual class this function is intended for. I think mock should be used when you don't want to commit to enough of a refactor to make your code unit-testable. 

In this case it'll make it clear what data this function is intended for. Or does simply instantiating the class have side-effects?

Reply:
No, I don't think instantiating it would have side-effects. Because you said "would then need half as much data to set up" in your comment above, it got into my head that I should try to make the tests create as little data as possible.

Reply:
haha cool - sorry :)

Reply:
I think I meant you don't need to set up multiple houses and persist to DB during the test setup